packages/components/src/internal/components/domainproperties/DomainForm.tsx

```diff
@@ -388,28 +427,149 @@ export class DomainFormImpl extends React.PureComponent<IDomainFormInput, IDomai
+
+        const updatedDomain = domain.merge({
+            fields: toggledFields,
+        }) as DomainDesign;
+        this.onDomainChange(updatedDomain, true);
+        this.setState((state) => ({selectAll: !state.selectAll, visibleSelection: newVisibleSelection}));
+    };
+
+    clearAllSelection = () => {
+        const { domain } = this.props;
+        const fields = domain.fields.map((field) => {
+            return field.set('selected', false);
+        });
+        const updatedDomain = domain.merge({fields}) as DomainDesign;
+        this.onDomainChange(updatedDomain, true);
+        this.setState({selectAll: false, visibleSelection: new Set()});
+    }
+
     onExportFields = () => {
         const { domain } = this.props;
+        const { visibleSelection } = this.state;
         const fields = domain.fields;
-        const filteredFields = fields.filter((field: DomainField) => field.visible);
+        let filteredFields = fields.filter((field: DomainField) => field.visible);
+        // Respect selection, if any selection exists
+        filteredFields = visibleSelection.size > 0
+            ? filteredFields.filter((field: DomainField) => field.selected)
+            : filteredFields;
+
         const fieldData = filteredFields.map(field => DomainField.serialize(field, false)).toArray();
         const fieldsJson = JSON.stringify(fieldData, null, 4);
 
         downloadJsonFile(fieldsJson, generateNameWithTimestamp('Fields') + '.fields.json');
     };
 
+    renderBulkFieldDeleteConfirm = () => {
+        const { domain } = this.props;
+        const { bulkDeleteConfirmInfo } = this.state;
+        const undeletableNames = bulkDeleteConfirmInfo.undeletableFields.map((i) => {return domain.fields.get(i).name});
+        const { howManyDeleted, undeletableWarning } = generateBulkDeleteWarning(bulkDeleteConfirmInfo, undeletableNames);
+
+        const thisFieldPlural = bulkDeleteConfirmInfo.deletableSelectedFields.length > 1 ? 'these fields' : 'this field';
+
+        return (
+            <ConfirmModal
+                title="Confirm Delete Selected Fields"
+                msg={
+                    <div>
+                        <p> {howManyDeleted} will be deleted. </p>
+                        <p> {undeletableWarning} </p>
+                        <p> Are you sure you want to delete {thisFieldPlural}? All of the related field data will also be deleted. </p>
+                    </div>
+                }
+                onConfirm={this.onBulkDeleteConfirm}
+                onCancel={this.onConfirmBulkCancel}
+                confirmVariant="danger"
+                confirmButtonText="Yes, Delete Fields"
+                cancelButtonText="Cancel"
+            />
+        );
+    }
+
+    onBulkDeleteClick = () => {
+        const { domain } = this.props;
+        const { visibleSelection } = this.state;
+        let fields = domain.fields;
+
+        let undeletableFields = [];
+        let deletableSelectedFields = [];
```

Comment:
These can all be `const` instead of `let`

Reply:
Oops! Thanks, fixed